### DISCUSSION TITLE
page style for extra sized page

### DISCUSSION THREAD

**Discussion by PiotrPanasiuk** [kjambunathan/org-mode-ox-odt#270] [Q&A]
Hi, for last few days I'm looking for solution to insert a3 wide page size with svg picture inside to the document where other pages in the document has a4 page size. During searching for potential solution in the internet I have met on mailing list an old solution which I could use:

`#+ATTR_ODT: :style "OrgPageBreakLandscape"`
`#+ATTR_ODT: :style "a3wide"`
`#+PAGEBREAK:`

`#+ATTR_ODT: :width 5 :style "OrgPageImage" :anchor "page"`

`#+BEGIN_TEXTBOX`
`| a          | b          |`
`| e          | f          |`
`#+END_TEXTBOX`

`#+ATTR_ODT: :style "OrgPageBreakDefault"`
`#+PAGEBREAK:`

in that manner(I hope) i have created an ott file with a3wide page style:
`<style:master-page style:name="a3wide" style:page-layout-name="Mpm8" draw:style-name="Mdp1">`
 after that I have adapted my org file to use it, but for unknown reason ox-odt exporter do not apply it during export.  
Now my only solution is to manually edit exported odt file to change pages size.
Is there a possibility to configure export for such page size from inside org file?

**Comment by kjambunathan** [comment]
**_Get a local copy of the recipe (and the exported result) by DOWNLOAD-ing the following ZIP  file_**

**[discussion 270 and issue 271: mix-of-a4-and-a3s.zip](https://github.com/kjambunathan/org-mode-ox-odt/files/14635132/discussion.270.and.issue.271.mix-of-a4-and-a3s.zip)**

-------


To get this

![mix-of-a3-and-a4-sized-pages](https://github.com/kjambunathan/org-mode-ox-odt/assets/19869864/11c2dc3b-0cb8-43a7-98a4-27249ac87568)

export this

```
#+ODT_PREFERRED_OUTPUT_FORMAT:  pdf

This is an a4 page

# #+ATTR_ODT: :width 20
#+ATTR_ODT: :page-style "A3PortraitPage" :style "OrgPageImage"
#+ATTR_ODT: :outer-frame (style "OrgPageImage")
#+ATTR_ODT: :scale 1.5
#+CAPTION: This is an a3 portrait page
[[./Sine.svg]]

#+ATTR_ODT: :page-style "Standard"
This is an a4 portrait page.

#+ATTR_ODT: :target "extra_automatic_styles" :backends (odt)
#+begin_src nxml
<style:page-layout style:name="A3PortraitLayout"
                   style:page-usage="mirrored">
  <style:page-layout-properties fo:margin-bottom="2cm"
				fo:margin-left="2cm"
				fo:margin-right="2cm"
				fo:margin-top="2cm"
				fo:page-height="42cm"
				fo:page-width="29.7cm"
				style:footnote-max-height="0cm"
				style:layout-grid-base-height="0.706cm"
				style:layout-grid-color="#c0c0c0"
				style:layout-grid-display="false"
				style:layout-grid-lines="20"
				style:layout-grid-mode="none"
				style:layout-grid-print="false"
				style:layout-grid-ruby-below="false"
				style:layout-grid-ruby-height="0.353cm"
				style:num-format="1"
				style:print-orientation="portrait"
				style:writing-mode="lr-tb">
    <style:footnote-sep style:adjustment="left"
			style:color="#000000"
			style:distance-after-sep="0.101cm"
			style:distance-before-sep="0.101cm"
			style:line-style="solid"
			style:rel-width="25%"
			style:width="0.018cm" />
  </style:page-layout-properties>
  <style:header-style />
  <style:footer-style>
    <style:header-footer-properties fo:margin-left="0cm"
                                    fo:margin-right="0cm"
                                    fo:margin-top="0.499cm"
                                    fo:min-height="0.6cm"
                                    style:dynamic-spacing="false" />
  </style:footer-style>
</style:page-layout>
#+end_src

#+ATTR_ODT: :target "master_styles" :backends (odt)
#+begin_src nxml
<style:master-page style:name="A3PortraitPage"
                   style:page-layout-name="A3PortraitLayout">
  <style:footer>
    <text:p text:style-name="OrgFooter">
      <text:page-number text:select-page="current">1</text:page-number>
    </text:p>
  </style:footer>
</style:master-page>
#+end_src



```

**Comment by kjambunathan** [comment]
> https://github.com/kjambunathan/org-mode-ox-odt/files/14635132/discussion.270.and.issue.271.mix-of-a4-and-a3s.zip

When you try the above recipe, I am expecting that you will flag some issues ... 

In anticipation of your "further" requests, I have opened [Issue#271: Pre-install commonly other commonly used page styles (a3, a5 etc). Also how do I insert a A3-sized page (with an image in it) in pre-dominantly A4-sized document](https://github.com/kjambunathan/org-mode-ox-odt/issues/271).  

If you have questions, you can raise it in this discussions thread, and if you identify further issues with recipe (="minor FR") you can register your requests as part of https://github.com/kjambunathan/org-mode-ox-odt/issues/271

**Reply by PiotrPanasiuk** [reply]
Sorry for delay, I'm not sure if I'm working with correct ox-odt version which is 9.6.620, however if yes, I have tried applied this solution with no success, I have many approaches like
1. When I generate output with Your template output is correct
2. When I choose the style name from style.xml there is no page size change in the output document
3. When I paste style from my ott file there I do not have page as extra sized, it is still normal size.
4. When I paste page style from Your template there is not page size in the output document, however at the same time I use ott template file which i use for my document.
and didn't know what is going on, after all of this I have started cut out part by part my document and now, I see strange thing, when I include in document table of contents:
`#+TOC: headlines 4`
`#+TOC: figures`
`#+TOC: listings`
`#+TOC: tables`
then from output/exported document, the style with oversized page disappears.

**Comment by PiotrPanasiuk** [comment]
After some additional tests I believe, I have managed „correct” oversized page injection to the document and actually I have one thing left, which bothers me, it is the picture size which is not updated according to dedicated features: 
`#+ATTR_ODT: :width 20`
or
 `#+ATTR_ODT: :scale 1.5`

so now, the thing which works for me looks like this:

`# +ATTR_ODT: :width 20`
`#+ATTR_ODT: :page-style "A3PortraitPage"`
`#+ATTR_ODT: :outer-frame (style "OrgPageImage")`
`#+ATTR_ODT: :scale 1.5`
`#+CAPTION: This is an a3 portrait page`
`[[./Sine.svg]]`

`#+ATTR_ODT: :page-style "Standard"`
`\\`
